docs/guides/tfhers.md

```diff
@@ -0,0 +1,242 @@
+    pbs_base_log=15,
+    pbs_level=2,
+    lwe_noise_distribution=0,  # this won't be needed so we set it to 0
+    glwe_noise_distribution=2.168404344971009e-19,
+    encryption_key_choice=tfhers.EncryptionKeyChoice.BIG,
+)
+# creating a TFHE-rs ciphertext type with crypto and encoding params
+tfhers_type = tfhers.TFHERSIntegerType(
+    is_signed=False,
+    bit_width=8,
+    carry_width=3,
+    msg_width=2,
+    params=tfhers_params,
+)
+# this partial will help us create TFHERSInteger with the given type instead of calling
+# tfhers.TFHERSInteger(tfhers_type, value) every time
+tfhers_int = partial(tfhers.TFHERSInteger, tfhers_type)
+```
+
+### Defining the Circuit and Compiling
+
+We will now define a simple addition function. This function takes TFHE-rs inputs, converts them to native (Concrete), runs a computation, and then converts them back to TFHE-rs. The circuit below is a common example that takes and produces TFHE-rs ciphertexts. However, there are other scenarios where you might not convert back to TFHE-rs, or you might convert to a different type than the input. Another possibility is to take one native ciphertext and one TFHE-rs ciphertext.
+
+```python
+
+def compute(x, y):
+    ####### TFHE-rs to Concrete #########
+    x = tfhers.to_native(x)  # the tfhers type will be derived from the x and y values
+    y = tfhers.to_native(y)
+    ####### TFHE-rs to Concrete #########
+
+    ####### Concrete Computation ########
+    res = x + y
+    ####### Concrete Computation ########
+
+    ####### Concrete to TFHE-rs #########
+    res = tfhers.from_native(res, tfhers_type)  # we have to specify the type we want to convert to
+    ####### Concrete to TFHE-rs #########
+    return res
+
+```
+
+We can compile the circuit as usual. The inputset is not really important here, as the compiler will consider the TFHE-rs bitwidth instead of trying to infer it from the inputset.
+
+```python
+
+compiler = fhe.Compiler(compute, {"x": "encrypted", "y": "encrypted"})
+inputset = [(tfhers_int(7), tfhers_int(8))]
+circuit = compiler.compile(inputset)
+
+```
+
+### The Bridge!
+
+We are going to create a TFHE-rs bridge that will assist in moving ciphertexts and keys between the two sides. You will need to define the input and output types of the circuit. This can be done for each individual input using a list, or globally for all inputs using a single value.
+
+```python
+tfhers_bridge = tfhers.new_bridge(circuit=circuit, input_types=tfhers_type, output_types=tfhers_type)
+```
+
+### Key Generation
+
+In order to establish a shared secret key between Concrete and TFHE-rs, there are two possible methods for key generation. The first method involves generating the Concrete keyset first and then using the shared secret key in TFHE-rs to partially generate the TFHE-rs keyset (i.e. starting from a secret key and generating the rest of the keys). The second method involves doing the opposite.
+
+**Note:** you should only perform one of two keygen scenario
```

Comment:
I would replace "Note: you should only perform one of two keygen scenario" by something like "Remember that one key generation is a partial key generation, to be sure that there is a unique and common secret key"